roofit/roofit/CMakeLists.txt

```diff
@@ -155,22 +155,28 @@ if (roofit-arch-libs)
   ROOT_LINKER_LIBRARY(RooFitCompute_SSE4.1  src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
   ROOT_LINKER_LIBRARY(RooFitCompute_AVX     src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
   ROOT_LINKER_LIBRARY(RooFitCompute_AVX2    src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
-  ROOT_LINKER_LIBRARY(RooFitCompute_AVX512f src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
-
+  
   # Add -fno-signaling-nans, -fno-trapping-math and -O3 to enable autovectorization
   # of the RooFitComputeLibrary (especially needed for GCC).
   set(common-flags $<$<OR:$<CONFIG:Release>,$<CONFIG:RelWithDebInfo>>:$<$<CXX_COMPILER_ID:GNU>:-fno-signaling-nans> -fno-trapping-math -O3>)
   target_compile_options(RooFitCompute_SSE4.1  PRIVATE ${common-flags} -msse4    -DRF_ARCH=SSE4)
   target_compile_options(RooFitCompute_AVX     PRIVATE ${common-flags} -mavx     -DRF_ARCH=AVX )
   target_compile_options(RooFitCompute_AVX2    PRIVATE ${common-flags} -mavx2    -DRF_ARCH=AVX2)
-  target_compile_options(RooFitCompute_AVX512f PRIVATE ${common-flags} -mavx512f -DRF_ARCH=AVX512F)
-
 
   if(vdt AND NOT builtin_vdt)
     target_include_directories(RooFitCompute_SSE4.1    PRIVATE ${VDT_INCLUDE_DIR})
-    target_include_directories(RooFitCompute_AVX2      PRIVATE ${VDT_INCLUDE_DIR})
     target_include_directories(RooFitCompute_AVX       PRIVATE ${VDT_INCLUDE_DIR})
-    target_include_directories(RooFitCompute_AVX512f   PRIVATE ${VDT_INCLUDE_DIR})
+    target_include_directories(RooFitCompute_AVX2      PRIVATE ${VDT_INCLUDE_DIR})
+  endif()
+
+  # AVX512 is only supported in gcc 6+
+  # All comercially available AVX512 capable processors support at least the skylake-avx512 instruction sets.
```

Comment:
Knights landing doesn't if I recall correctly, but you can just say that we are only targeting skylake+.
`comMercially`